android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -92,7 +97,13 @@ constructor(
     data: Bundle = Bundle(),
     generateCarePlanWithWorkflowApi: Boolean = false,
   ): CarePlan? {
-    val planDefinition = defaultRepository.loadResource<PlanDefinition>(planDefinitionId)
+    val planDefinition =
+      if (appIdExistsAndIsNotNull(sharedPreferencesHelper)) {
+        configurationRegistry.retrieveResourceFromConfigMap<PlanDefinition>(
+          resourceId = planDefinitionId,
+        )
+      } else defaultRepository.loadResource<PlanDefinition>(planDefinitionId)
```

Comment:
The expectation was such that the configs should be added to the database. This code refactor was not required. Save the configs to the DB and revert these.